src/sqlfluff/core/linter/linter.py

```diff
@@ -472,6 +473,14 @@ def lint_fix_parsed(
         """Lint and optionally fix a tree object."""
         # Keep track of the linting errors
         all_linting_errors = []
+        # Keep track of the linting errors on the very first linter pass. The
+        # list of issues output by "lint" and "fix" only includes issues present
+        # in the initial SQL code, EXCLUDING ANY ISSUES THAT MAY BE CREATED BY
+        # THE FIXES THEMSELVES.
+        # The value is initially None, and is set to a list value on the first
+        # pass. The value None is used as a sentinel to indicate we're on the
+        # first pass.
+        initial_linting_errors = None
```

Comment:
@tunetheweb: I added this code and comment to make the code clearer. This allowed me to update two places that were doing `if phase == "main" and loop == 0:` with `if initial_linting_errors is None:`. I think this clarifies the behavior. The type checker in PyCharm (not mypy itself, though) has always complained that `initial_linting_errors` was not guaranteed to be assigned a value, and this makes it happier too.